--- scorealign/Makefile.linux.orig	2013-11-09 08:42:50.000000000 +0100
+++ scorealign/Makefile.linux	2013-11-09 12:08:54.994329429 +0100
@@ -25,14 +25,12 @@
 c++ = g++ $(c_opts)
 cclink = gcc
 c++link = g++
-syslibs = -lm
+syslibs = -lm -lportSMF -lsndfile
 
-scorealign_o = comp_chroma.o gen_chroma.o scorealign.o snd/audionone.o \
-               snd/sndcvt.o snd/sndheader.o snd/sndio.o snd/snd.o \
-               snd/ieeecvt.o snd/sndlinux.o snd/sndfaillinux.o \
-               fft3/FFT.o $(PORTSMF)/mfmidi.o $(PORTSMF)/allegro.o \
-               $(PORTSMF)/allegrosmfrd.o $(PORTSMF)/allegrosmfwr.o \
-               $(PORTSMF)/allegrord.o $(PORTSMF)/strparse.o trace.o \
+scorealign_o = comp_chroma.o gen_chroma.o scorealign.o \
+               audiofilereader.o audioreader.o alignfiles.o main.o \
+               fft3/FFT3.o \
+               trace.o \
                regression.o sautils.o curvefit.o hillclimb.o
 
 .SUFFIXES: .cpp .c .h .aur .o
